runtime/lib/parser/yaml/sfYamlInline.php

```diff
@@ -127,7 +125,7 @@ protected static function dumpArray($value)
     if (
       (1 == count($keys) && '0' == $keys[0])
       ||
-      (count($keys) > 1 && array_reduce($keys, create_function('$v,$w', 'return (integer) $v + $w;'), 0) == count($keys) * (count($keys) - 1) / 2))
+      (count($keys) > 1 && array_reduce($keys, function($v, $w) { return (integer) $v + $w; }, 0) == count($keys) * (count($keys) - 1) / 2))
```

Comment:
Starting from PHP 7.1 this causes warnings when array key stored in $w variable is string instead of integer. I think we could easily replace whole if() condition with 

`if (array_values($value) === $value)`

Reply:
I didn't get this warning in the phpunit, but what I did when applied the change now is:

`There were 2 failures:

1) PropelYAMLParserTest::testFromArray with data set #0 (array(), '{  }', 'empty array')
PropelYAMLParser::fromArray() converts from empty array correctly
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'{  }'
+'[]'

/usr/src/app/test/testsuite/runtime/parser/PropelYAMLParserTest.php:85

2) PropelYAMLParserTest::testToYAML with data set #0 (array(), '{  }', 'empty array')
PropelYAMLParser::toYAML() converts from empty array correctly
Failed asserting that two strings are equal.
--- Expected
+++ Actual
@@ @@
-'{  }'
+'[]'`

Which looks like some edge cases the original implementation was covering

Reply:
The reason why you didn't get warning "A non-numeric value encountered" is probably your error_reporting setting then.
From YAML-Array convertion perspective there's no difference between '{ }' and '[] for an empty array, so it would be OK to tweak up unit tests too.

Reply:
You could also change condition to:

`if (count($value) > 0 && array_values($value) === $value)`

and leave unit tests as it is